features/datachoices/src/main/java/org/opennms/features/datachoices/internal/usagestatistics/UsageStatisticsReporter.java

```diff
@@ -260,6 +269,19 @@ public UsageStatisticsReportDTO generateReport() {
             LOG.warn("An error occurred while retrieving the system id. " +
                         "The usage report will be submitted with a null system id.", e);
         }
+
+
+        try {
+            final long currentTime = System.currentTimeMillis();
+            final long last24Hours = currentTime - Duration.ofHours(24).toMillis();
+
+            final List<Filter> filters = Collections.singletonList(new TimeRangeFilter(last24Hours, currentTime));
+            final long flowCount = flowQueryService.getFlowCount(filters).get();
+            usageStatisticsReport.setFlowCountPerSecond(flowCount);
+        } catch (InterruptedException | ExecutionException e) {
+            LOG.warn("An error occurred while retrieving the flow count. ", e);
+        }
```

Comment:
Move, all of this new code block into separate method.

Check if `flowQueryService` is not null before accessing.